obs-studio-server/source/nodeobs_api.cpp

```diff
@@ -622,6 +627,129 @@ void OBS_API::OBS_API_getPerformanceStatistics(
+			    currentHotkeyInfo.objectName = obs_source_get_name(key_source);
+			    currentHotkeyInfo.objectType = OBS_HOTKEY_REGISTERER_SOURCE;
+			    break;
+		    }
+		    case OBS_HOTKEY_REGISTERER_OUTPUT: {
+			    auto* weak_output            = static_cast<obs_weak_output_t*>(registerer);
+			    auto  key_output             = OBSGetStrongRef(weak_output);
+			    currentHotkeyInfo.objectName = obs_output_get_name(key_output);
+			    currentHotkeyInfo.objectType = OBS_HOTKEY_REGISTERER_OUTPUT;
+			    break;
+		    }
+		    case OBS_HOTKEY_REGISTERER_ENCODER: {
+			    auto* weak_encoder           = static_cast<obs_weak_encoder_t*>(registerer);
+			    auto  key_encoder            = OBSGetStrongRef(weak_encoder);
+			    currentHotkeyInfo.objectName = obs_encoder_get_name(key_encoder);
+			    currentHotkeyInfo.objectType = OBS_HOTKEY_REGISTERER_ENCODER;
+			    break;
+		    }
+		    case OBS_HOTKEY_REGISTERER_SERVICE: {
+			    auto* weak_service           = static_cast<obs_weak_service_t*>(registerer);
+			    auto  key_service            = OBSGetStrongRef(weak_service);
+			    currentHotkeyInfo.objectName = obs_service_get_name(key_service);
+			    currentHotkeyInfo.objectType = OBS_HOTKEY_REGISTERER_SERVICE;
+			    break;
+		    }
+		    }
+
+		    // Key defs
+		    auto       key_name = std::string(obs_hotkey_get_name(key));
+		    auto       desc     = std::string(obs_hotkey_get_description(key));
+		    const auto hotkeyId = obs_hotkey_get_id(key);
+
+		    // Parse the key name and the description
+		    key_name = key_name.substr(key_name.find_first_of(".") + 1);
+		    std::replace(key_name.begin(), key_name.end(), '-', '_');
+		    std::transform(key_name.begin(), key_name.end(), key_name.begin(), ::toupper);
+		    std::replace(desc.begin(), desc.end(), '-', ' ');
+		    desc = ToTitle(desc);
+
+		    currentHotkeyInfo.hotkeyName = key_name;
+		    currentHotkeyInfo.hotkeyDesc = desc;
+		    currentHotkeyInfo.hotkeyId   = hotkeyId;
+		    hotkeyInfos.push_back(currentHotkeyInfo);
+
+		    return true;
+	    },
+	    &hotkeyInfos);
+
+	rval.push_back(ipc::value((uint64_t)ErrorCode::Ok));
+
+	// For each hotkey that we've found
+	for (auto& hotkeyInfo : hotkeyInfos) {
+		rval.push_back(ipc::value(hotkeyInfo.objectName));
+		rval.push_back(ipc::value(uint32_t(hotkeyInfo.objectType)));
+		rval.push_back(ipc::value(hotkeyInfo.hotkeyName));
+		rval.push_back(ipc::value(hotkeyInfo.hotkeyDesc));
+		rval.push_back(ipc::value(uint64_t(hotkeyInfo.hotkeyId)));
+	}
+
+	AUTO_DEBUG;
+}
+
+void OBS_API::ProcessHotkeyStatus(
+    void*                          data,
+    const int64_t                  id,
+    const std::vector<ipc::value>& args,
+    std::vector<ipc::value>&       rval)
+{
+	obs_hotkey_id hotkeyId = args[0].value_union.ui64;
+	uint64_t      press    = args[1].value_union.i32;
+
+	// TODO: Check if the hotkey ID is valid
+	obs_hotkey_enable_callback_rerouting(true);
```

Comment:
Do we need to set that to false at some point or obs does it when the callback is done with its work?